get_http.go

```diff
@@ -117,13 +118,54 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 		g.Client = httpClient
 	}
 
-	resp, err := g.Client.Get(u.String())
+	// check to see whether user has specified a range of bytes to download.
+	// if user has, but the range is invalid, fall back to downloading the
+	// whole file
+	isPartialDownload := false
+	byte_range, err := GetByteRange(u)
+	if err != nil {
+		if strings.Compare("No byte range provided", err.Error()) != 0 {
+			fmt.Printf("%s; going to download entire file without a range request",
+				err)
+		}
+	}
+
+	req, err := http.NewRequest("HEAD", u.String(), nil)
+
+	if len(byte_range) == 2 {
```

Comment:
might make sense to have `GetByteRange` return an error if it didn't find 2 numbers?

Reply:
already does; I guess I could just use this error instead of re-checking though.

Reply:
ah, I see, this seems fine then. Could probably combine this code with the err checking above, but that's also trivial